plugin/jacoco_coverage_metrics.go

```diff
@@ -0,0 +1,138 @@
+package plugin
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io"
+	"log"
+	"os"
+)
+
+type Report struct {
+	XMLName  xml.Name  `xml:"report"`
+	Counters []Counter `xml:"counter"`
+	Packages []Package `xml:"package"`
+}
+
+type Counter struct {
+	Type    string `xml:"type,attr"`
+	Missed  int    `xml:"missed,attr"`
+	Covered int    `xml:"covered,attr"`
+}
+
+type Package struct {
+	Name     string    `xml:"name,attr"`
+	Counters []Counter `xml:"counter"`
+}
+
+type JacocoCoverageThresholds struct {
+	InstructionCoverageThreshold string
+	BranchCoverageThreshold      string
+	LineCoverageThreshold        string
+	ComplexityCoverageThreshold  int
+	MethodCoverageThreshold      string
+	ClassCoverageThreshold       string
+}
+
+type JacocoCoverageThresholdsValues struct {
+	InstructionCoverageThreshold float64
+	BranchCoverageThreshold      float64
+	LineCoverageThreshold        float64
+	ComplexityCoverageThreshold  int
+	MethodCoverageThreshold      float64
+	ClassCoverageThreshold       float64
+}
+
+func (j *JacocoCoverageThresholds) ToFloat64() JacocoCoverageThresholdsValues {
+	return JacocoCoverageThresholdsValues{
+		InstructionCoverageThreshold: ParsePercentage(j.InstructionCoverageThreshold),
+		BranchCoverageThreshold:      ParsePercentage(j.BranchCoverageThreshold),
+		LineCoverageThreshold:        ParsePercentage(j.LineCoverageThreshold),
+		ComplexityCoverageThreshold:  j.ComplexityCoverageThreshold,
+		MethodCoverageThreshold:      ParsePercentage(j.MethodCoverageThreshold),
+		ClassCoverageThreshold:       ParsePercentage(j.ClassCoverageThreshold),
+	}
+}
+
+func ParsePercentage(percentage string) float64 {
+	var value float64
+	_, err := fmt.Sscanf(percentage, "%f", &value)
+	if err != nil {
+		log.Fatalf("Error parsing percentage: %v", err)
```

Comment:
could you please replace the usage of `log` with `logrus` to adhere to the plugin dev guidelines?

Reply:
fixed all log. usages to logrus.us